### DISCUSSION TITLE
Forge compile error

### DISCUSSION THREAD

**Discussion by Edacheclarus2** [Cyfrin/foundry-full-course-cu#344] [General]
Dear all, please why do i get a "Nothing to compile" reply when i run forge compile?

**Comment by shikhar229169** [comment]
It may happen because of any of the following reasons:
- Contracts were already compiled and you never modified them after, so when you run compile at this stage it will say Nothing to Compile
- Contracts are not in src folder.

**Reply by bh1636** [reply]
same, i copy paste simplestorage from github and opened a new file .named it as simplestorage.sol under src folder, then from terminal wrote just forge build and it gives 'nothing to compile'

**Reply by Edacheclarus2** [reply]
Yh, I think it was due to the first reason. 
I deleted and force forged it again and now it works just fine.

Thank you for the reply

**Reply by bh1636** [reply]
thanks, this answer was little for you but too big for me in order to be a solidity developer :D

**Reply by Edacheclarus2** [reply]
Were you able to sort it, or what challenge are you still facing?

**Reply by bh1636** [reply]
yes thank you

**Reply by Airikr02** [reply]
Hello, 
I'm encountering the same issue here. Could you please share how you managed to make it work exactly? I attempted your solution, but I suspect I might have made a mistake somewhere along the line...
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/78366349/1dd1d1fe-9adf-4b2d-b5b5-120bdcffff39)
